### ISSUE TITLE
Questions about inpainting

### ISSUE THREAD

**Issue by Polaris0421** [CreamyLong/stable-diffusion#8]
Hey, thanks for this helpful repo and I have starred it. But I have some questions which are not clear in README.md:
1.How can I use pre-trained autoencoder model? I didn't see where to combine the downloaded pre-trained autoencoder model in the second stage training process. Actually the same problem with the train-from-scratch autoencoder model.
2.I didn't find the train.yaml and inpainting.py files mentioned in the 'Inpainting' part of README.md,  and I'm not sure how to resume training after download the wights. Can you be more specific in your README.md?
3.You have several inpainting_xx in your script folder, what are they?
I'll really appreciate it if you can help me with those questions. Your repo is the only one I find that is likely still working on the inpainting task with ldm.

**Comment by Polaris0421** [comment]
Also, the path 'models/ldm/inpainting_big/last.ckpt' you point to for downloading the wights does not exist. Where should I put last.ckpt?

**Comment by CreamyLong** [comment]
1. In the readme, you can download the pre-trained autoencoder model [here](https://github.com/CreamyLong/stable-diffusion?tab=readme-ov-file#pretrained-autoencoding-models).  For training the autoencoder model from scratch, have a look at [this](https://github.com/CreamyLong/stable-diffusion?tab=readme-ov-file#training-autoencoder-models)
2. I have corrected the name of Yaml. the inpainting model is downloaded [here](https://github.com/CreamyLong/stable-diffusion/blob/master/scripts/download/download_models.sh)
3. All are for inference, just for different purposes
4. same as 2

**Comment by Polaris0421** [comment]
Thanks again for your clear and structured response.  I'm still confused about the first question. I checked the 'inpainting_example_overfit.yaml' and related classes, and I didn't find the use of pre-trained autoencoder model anywhere. So this means actually training ldm does require a pre-trained autoencoder model? 
My project is inpainting task on one channel grayscale Image and I've already trained a VQ autoencoder from scratch, but now I',m not sure how to use it.

**Comment by Polaris0421** [comment]
I get more confused as there is an annotation ' NO TRAIN FOR AUTOENCODER' in the 'inpainting_example_overfit.yaml'. I find the parameter'ckpt_path' in class VQModel, and I wonder if it's necessary to  set the path to my pre_trained VQmodel?

**Comment by CreamyLong** [comment]
> re. So this means actually training ldm does require a pre-trained autoencoder model?

Yes, pre-trained autoencoder model is needed. you can learn how to load the pre-trained model [here](https://github.com/CreamyLong/stable-diffusion/blob/1def3a9178dcf05a41acd69bc39ddf6af8e1b4b4/configs/latent-diffusion/unconditional/lsun_bedrooms-ldm-vq-4.yaml#L39C1-L40C1). By the way, stable-diffusion is a very complex AI model, I recommended you read the paper carefully first which will helps you understand the code better.

**Comment by Polaris0421** [comment]
Thanks for your suggestion. I've successfully began training the inpainting model on the 100 256*256 one channel grayscale images, and my batch size is 12. After 270 epochs, the result is still bad. My autoencoder works fine and I'm not sure how many epochs I need. I think the author didn't mention the number of epochs they trained the inpainting model, may I ask your related settings(batch size, hardware, epochs etc) for training the inpainting model?

**Comment by CreamyLong** [comment]
> Thanks for your suggestion. I've successfully began training the inpainting model on the 100 256*256 one channel grayscale images, and my batch size is 12. After 270 epochs, the result is still bad. My autoencoder works fine and I'm not sure how many epochs I need. I think the author didn't mention the number of epochs they trained the inpainting model, may I ask your related settings(batch size, hardware, epochs etc) for training the inpainting model?

Sorry to hear that. Do you mind sharing the sample of your dataset and your result of inference?

**Comment by Polaris0421** [comment]
of course. This is my input origin image, some digital elevation data scaled to 0-255, and my autoencoder reconstructs fine: 
![ec3302c62add770c781d870b17b1d0c](https://github.com/CreamyLong/stable-diffusion/assets/160501296/b4d2e5a1-9839-4677-8609-f041fbaef2b4)
The following images are from image logger,2250 global steps and 434 epochs. This is my input mask image example:
and this is corresponding sample:
By the way, the sample result from image logger should be the same with another inference procedure, right?

**Comment by CreamyLong** [comment]
This looks like the training process at least works. I assume that the diffusion model was not trained very well. How many samples did you use for training? Also, you only use the grayscale images so that the number of channels is 1, do you change the number of channels in unet?

**Comment by Polaris0421** [comment]
- I just use 100 samples because I wanna figure out if the model works first. I only have two 3090s to use, and I think it's too slow to use more training samples. (rightnow it takes me 12h to finish around 80 epochs)
- I didn't change the number of channels in Unet as my latent embedding in autoencoder is 64*64*3, the number of channels for Unet does not change. I just change the size in Unet and in&out channel number in autoencoder of the config.
- Maybe I should train more epochs? The loss seems to still decease(but of very little magnitude)

**Comment by CreamyLong** [comment]
In my experience, 10k images are the least for inpainting model if your pictures are not very complicated. Otherwise, overfitting will happen.  I recommend you do more experiments because it depends on the datasets and model structure.

**Comment by Polaris0421** [comment]
thanks, overfitting is ok for me right now as long as it shows good inpainting ability. I'll let you know if I get satisfied results. Thanks again!

**Comment by Polaris0421** [comment]
After more epochs the result becomes good. Thx for your help!

**Comment by IwantNewway** [comment]
> After more epochs the result becomes good. Thx for your help!

could u tell me how  train in your one channel grayscale images, such as yaml ? thank u
